docker-vertica-v2/Dockerfile

```diff
@@ -16,26 +16,35 @@ ARG BASE_OS_VERSION
 ARG BUILDER_OS_NAME
 ARG BUILDER_OS_VERSION
 ARG MINIMAL=""
-FROM ${BUILDER_OS_NAME}:${BUILDER_OS_VERSION} as builder
+FROM ${BUILDER_OS_NAME}:${BUILDER_OS_VERSION} AS builder
 
-ARG VERTICA_RPM
+ARG VERTICA_X86_RPM
+ARG VERTICA_ARM64_RPM
 ARG MINIMAL
 
-COPY ./packages/${VERTICA_RPM} /tmp/
+COPY ./packages/*.rpm /tmp/
 # this is a script which removes unnecessary stuff from the
 # container image
 COPY ./packages/cleanup.sh /tmp/
 COPY ./packages/package-checksum-patcher.py /tmp/
 COPY ./packages/httpstls.json /tmp/
 
 SHELL ["/bin/bash", "-o", "pipefail", "-c"]
+# Update is needed to be confident that we're picking up
+# fixed libraries.
+# Using --nobest to make it easier yum install to work. This stage isn't used
+# for the final image, so any package is good enough. We just need to install
+# the vertica rpm and copy that over.
+
+# It can be challenging to identify the exact architecture that the build file is targetting when doing a multi-platform build
+# The pain is described in https://github.com/BretFisher/multi-platform-docker-build quite well
+# We only need to do this because we have separate vertica RPMs for ARM and x86. 
+# There is no need for us to differentiate between the vriants of ARM. So we consider those as equivalent for our RPM install.
+
+# Don't move the ARG from here. The ARG is only read by the build system if its declared after the SHELL
+ARG TARGETARCH
```

Comment:
How does the dockerfile receive the value of "TARGETARCH"? I didn't see it on build-arg list of the makefile below.

Reply:
It is a predefined arg that are set automatically. We do set the target platform with the --platform flag on docker build, docker pulls out the architecture component of TARGETPLATFORM (that is set by the --platform).